### DISCUSSION TITLE
Java no longer working?

### DISCUSSION THREAD

**Discussion by 12122234** [FabricMC/fabric-api#2532] [Mod Dev Support] [answered]
I was making a mod until next day java kept saying "The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted." and even when trying to launch the game it didn't work so what do i do?
Here is the log and another one in the comments:
`{
  message: "Use the JDK from '' as the initial default project JDK.",
  level: 'info',
  timestamp: '2022-09-15 09:20:06.470'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_ss_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\ss_ws',
  level: 'info',
  timestamp: '2022-09-15 09:20:12.083'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\jdt_ws',
  level: 'info',
  timestamp: '2022-09-15 09:20:15.155'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:21:07.109'
}
{
  message: '[Info  - 9:21:07 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:21:07.263'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:21:09.667'
}
{
  message: '[Info  - 9:21:09 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:21:09.667'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:21:12.355'
}
{
  message: '[Info  - 9:21:12 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:21:12.355'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:21:14.934'
}
{
  message: '[Info  - 9:21:14 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:21:14.935'
}
{
  message: 'The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted.',
  level: 'error',
  timestamp: '2022-09-15 09:21:18.704'
}
{
  message: '[Error - 9:21:18 AM] Connection to server got closed. Server will not be restarted.',
  level: 'info',
  timestamp: '2022-09-15 09:21:18.707'
}
{
  message: 'Starting Java server with: C:\\Program Files\\Java\\jdk-17.0.4.1\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.11.2022091504-win32-ia32\\lombok\\lombok-1.18.24.jar -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.11.2022091504-win32-ia32\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.11.2022091504\\config_ss_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\ss_ws',
  level: 'info',
  timestamp: '2022-09-15 09:24:41.415'
}
{
  message: 'Starting Java server with: C:\\Program Files\\Java\\jdk-17.0.4.1\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.11.2022091504-win32-ia32\\lombok\\lombok-1.18.24.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.11.2022091504-win32-ia32\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.11.2022091504\\config_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\jdt_ws',
  level: 'info',
  timestamp: '2022-09-15 09:24:41.918'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:24:44.698'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:24:45.593'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:26:07.182'
}
{
  message: '[Info  - 9:26:07 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:26:07.191'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:26:08.094'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:26:14.654'
}
{
  message: '[Info  - 9:26:14 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:26:14.655'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:26:15.655'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:26:20.416'
}
{
  message: '[Info  - 9:26:20 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:26:20.417'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:26:20.641'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:26:23.676'
}
{
  message: '[Info  - 9:26:23 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:26:23.677'
}
{
  message: 'WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign\r\n',
  level: 'info',
  timestamp: '2022-09-15 09:26:23.884'
}
{
  message: 'The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted.',
  level: 'error',
  timestamp: '2022-09-15 09:26:27.313'
}
{
  message: '[Error - 9:26:27 AM] Connection to server got closed. Server will not be restarted.',
  level: 'info',
  timestamp: '2022-09-15 09:26:27.314'
}
{
  message: "Use the JDK from '' as the initial default project JDK.",
  level: 'info',
  timestamp: '2022-09-15 09:47:44.345'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_ss_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\ss_ws',
  level: 'info',
  timestamp: '2022-09-15 09:47:45.308'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\jdt_ws',
  level: 'info',
  timestamp: '2022-09-15 09:47:47.188'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:49:17.146'
}
{
  message: '[Info  - 9:49:17 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:49:17.156'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:49:25.981'
}
{
  message: '[Info  - 9:49:25 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:49:25.981'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:49:31.581'
}
{
  message: '[Info  - 9:49:31 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:49:31.582'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:49:39.829'
}
{
  message: '[Info  - 9:49:39 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:49:39.830'
}
{
  message: 'The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted.',
  level: 'error',
  timestamp: '2022-09-15 09:49:44.481'
}
{
  message: '[Error - 9:49:44 AM] Connection to server got closed. Server will not be restarted.',
  level: 'info',
  timestamp: '2022-09-15 09:49:44.483'
}
{
  message: "Use the JDK from '' as the initial default project JDK.",
  level: 'info',
  timestamp: '2022-09-15 09:55:33.878'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_ss_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\ss_ws',
  level: 'info',
  timestamp: '2022-09-15 09:55:37.293'
}
{
  message: 'Starting Java server with: c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\jre\\17.0.4-win32-x86_64\\bin\\java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/sun.nio.fs=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Djava.import.generatesMetadataFilesAtProjectRoot=false -Dfile.encoding=utf8 -DwatchParentProcess=false -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -Xlog:jni+resolve=off -javaagent:c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\lombok\\lombok-1.18.24.jar -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java -jar c:\\Users\\mohay\\.vscode\\extensions\\redhat.java-1.10.0-win32-x64\\server\\plugins\\org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar -configuration c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\globalStorage\\redhat.java\\1.10.0\\config_win -data c:\\Users\\mohay\\AppData\\Roaming\\Code\\User\\workspaceStorage\\8d0116de6c8605dc71d954ca4a9db494\\redhat.java\\jdt_ws',
  level: 'info',
  timestamp: '2022-09-15 09:55:37.948'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:55:43.021'
}
{
  message: '[Info  - 9:55:43 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:55:43.030'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:55:48.139'
}
{
  message: '[Info  - 9:55:48 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:55:48.140'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:55:51.775'
}
{
  message: '[Info  - 9:55:51 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:55:51.776'
}
{
  message: 'The Language Support for Java server crashed and will restart.',
  level: 'error',
  timestamp: '2022-09-15 09:55:55.807'
}
{
  message: '[Info  - 9:55:55 AM] Connection to server got closed. Server will restart.',
  level: 'info',
  timestamp: '2022-09-15 09:55:55.809'
}
{
  message: 'The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted.',
  level: 'error',
  timestamp: '2022-09-15 09:55:59.437'
}
{
  message: '[Error - 9:55:59 AM] Connection to server got closed. Server will not be restarted.',
  level: 'info',
  timestamp: '2022-09-15 09:55:59.438'
}
`

**Comment by 12122234** [comment]
Sorry i can't put the other here it was too long and I was unable to send it :(

**Comment by warjort** [comment, marked as answer]
This is a vscode error. It says your language server plugin for java was stopped and can't be restarted.
````The Language Support for Java server crashed 5 times in the last 3 minutes. The server will not be restarted````
The language server is the thing that validates what you are typing as java code is correct.

Try restarting your ide. If that doesn't fix it, you will need to contact Redhat who make the plugin.
https://marketplace.visualstudio.com/items?itemName=redhat.java

**Reply by 12122234** [reply]
I uninstalled and reinstalled the extension and it now works fine!
I'll mark your suggestion as answer though

**Reply by 12122234** [reply]
Sadly, I have to rework my whole mod because all of the files were corrupted.
